frontend/lib/identification/activation_workflow/activation_code_scanner_page.dart

```diff
@@ -27,8 +34,27 @@ class ActivationCodeScannerPage extends StatelessWidget {
   Future<void> _onCodeScanned(BuildContext context, String code) async {
     final provider = Provider.of<ActivationCodeModel>(context, listen: false);
     Future<void> showError(String msg) async => QrParsingErrorDialog.showErrorDialog(context, msg);
+
+    final client = GraphQLProvider.of(context).value;
     try {
       final activationCode = const ActivationCodeParser().parseQrCodeContent(code);
+      final projectId = Configuration.of(context).projectId;
+      final generator = OTPGenerator(activationCode.totpSecret);
+      final otp = generator.generateOTP();
+      final verifyQrCode = DynamicVerifyCode(
+        info: activationCode.info,
+        pepper: activationCode.pepper,
+        otp: otp.code,
+      );
+      final valid = await queryServerVerification(client, projectId, verifyQrCode);
+      if (!valid) {
+        await showError(
+          "Die zu prüfende Karte konnte vom Server nicht "
+          "verifiziert werden!",
+        );
+        return;
+      }
+
       provider.setCode(activationCode);
```

Comment:
Not sure, if you already handle this: If the user has no internet connection, a "nicer" message like "Bitte prüfen Sie Ihre Internetverbindung" should be shown. Also, optimally, before scanning the activation code, the user should be hinted that this action needs an internet connection.

Reply:
This was basically copied from the verification workflow. I guess this rather checks whether the hash exists and such. So "Die prüfende Karte ist ungültig" would fit better.